processor/signoztailsampler/config.go

```diff
@@ -194,9 +136,40 @@ type SpanCountCfg struct {
 	MinSpans int32 `mapstructure:"min_spans"`
 }
 
+type PolicyFilterCfg struct {
+	// values: and | or
+	FilterOp string `mapstructure:"filter_op"`
+
+	StringAttributeCfgs  []StringAttributeCfg  `mapstructure:"string_attributes"`
+	NumericAttributeCfgs []NumericAttributeCfg `mapstructure:"numeric_attributes"`
+}
+
+// PolicyCfg identifies policy rules in policy group
+type PolicyCfg struct {
+	// name of the policy
+	Name string `mapstructure:"name"`
+
+	// Type of the policy this will be used to match the proper configuration of the policy.
+	Type PolicyType `mapstructure:"type"`
+
+	// Set to true for sampling rule (root) and false for conditions
+	Root bool `mapstructure:"root"`
+
+	Priority int `mapstructure:"priority"`
+
+	// sampling applied when  PolicyFilter matches
+	ProbabilisticCfg `mapstructure:"sampling"`
+
+	// filter to activate policy
+	PolicyFilterCfg `mapstructure:"policy_filter"`
+
+	SubPolicies []PolicyCfg `mapstructure:"sub_policies"`
```

Comment:
The sub-policies follow the same semantics of root policy. 

The approach is derived from prom alert manager routes and it was picked to support rule hierarchy. 
https://prometheus.io/docs/alerting/latest/configuration/#route

I don't see a reason to limit this feature to one level through architecture. It can be a choice at higher level - through config or other means. 

If understanding is issue, I will document the code in further detail.

Reply:
this is done. removed self-ref in policy